qiskit/circuit/quantumcircuit.py

```diff
@@ -1802,17 +1809,18 @@ def depth(self) -> int:
         # We treat barriers or snapshots different as
         # They are transpiler and simulator directives.
         # The max stack height is the circuit depth.
+        whitelist = list(filter(filter_function, self._data))
```

Comment:
since we only do lookups a set is probably a better choice

```suggestion
        whitelist = set(filter(filter_function, self._data))
```